### DISCUSSION TITLE
Coding with LLMs

### DISCUSSION THREAD

**Discussion by dominikpeter** [zauberzeug/nicegui#3783] [Ideas / Feature Requests]
Nowadays, everyone codes with AI in one way or another. Unfortunately, often the training data is too old or the AI did not have enough training data to work with `NiceGUI`.

To solve this problem I like the approach of `FastHTML`:

[Getting help from AI](https://github.com/AnswerDotAI/fasthtml?tab=readme-ov-file#getting-help-from-ai)
[Text File](https://docs.fastht.ml/llms-ctx.txt)

Something like this would be nice :-)  for `NiceGUI`, so you could use it in e.g. Claude Projects to help you coding `NiceGUI`applications.

**Comment by aersam** [comment]
Maybe something to derive mostly from the docs?

**Comment by falkoschindler** [comment]
We started something similar quite a while ago:
https://github.com/zauberzeug/nicegui/wiki/Chat-GPT-4-Prompt

Related discussion: https://github.com/zauberzeug/nicegui/discussions/1371

**Reply by alikalik9** [reply]
This is a start, but GPT still doesnt't know the Nicegui library and hallucinates most of the time. 
What works really well is Claude Sonnet 3.5. NiceGUI was definetly part of the training set, so it **doesn't** hallucinate most of the time.

**Comment by me21** [comment]
Cursor.sh has implemented custom doc parsing in their IDE, which is a fork of VS Code: https://www.cursor.com/features

> Reference popular libraries using @LibraryName, or add your own using @Docs → Add new doc.

Hopefully it can make responses more relevant if NiceGUI docs are included as part of the context. I haven't worked on a NiceGUI project there yet, but I've switched to it from vanilla VS Code + Copilot plugin for another project.
